policy-controller/src/main.rs

```diff
@@ -158,10 +158,21 @@ async fn main() -> Result<()> {
         kubert::index::namespaced(index.clone(), http_routes).instrument(info_span!("httproutes")),
     );
 
+    // Create the status controller lease and start trying to claim it.
+    let lease = status::index::create_lease_manager(runtime.client()).await?;
+    let pod_name =
+        std::env::var("HOSTNAME").expect("Failed to fetch `HOSTNAME` environment variable");
+    // todo: These should probably be configurable or set as consts
+    let params = kubert::lease::ClaimParams {
+        lease_duration: std::time::Duration::from_secs(30),
+        renew_grace_period: std::time::Duration::from_secs(1),
+    };
+    let (claims, _task) = lease.spawn(pod_name.clone(), params).await?;
```

Comment:
I might be wrong, but the way I understand `spawn` is that it will drive the future inside spawn to completion (i.e it will return Poll::Pending until it claims the lease). Once it has claimed the lease, it will give us a receiver and the bg task handle for the receive watch. As long as both of these stay in scope, the controller will maintain its lease.

My questions:

1. Does my understanding line up with yours? I might very well be wrong here.
2. If we can't claim the lease, we won't make any progress here? Wouldn't we still want the index to work and serve discovery requests? We would probably only want to block from doing writes and that's it.